unblob/pool.py

```diff
@@ -67,13 +89,16 @@ def __init__(
             )
             for _ in range(process_num)
         ]
+        self._should_stop_processing = False
+        _register_signal_handler()
 
     def start(self):
         for p in self._procs:
             p.start()
 
     def close(self):
-        self._input.join()
```

Comment:
After thinking about it a while, I think this join here is unsound: We either called `process_until_done` and let it do its job, then the `is_empty` call returning False is equivalent to `join` returning without blocking, or we didn't call `process_until_done` or it failed with an exception, in which case the queue never gets emptied and this line will block forever.

TL; DR we should never call `join`. Also, by removing that line, everything will terminate properly and we just need to silence the traceback printed by handling KeyboardInterrupt in the children.